thefederation/views.py

```diff
@@ -9,12 +9,14 @@
 
 
 def register_view(request, host):
+    json = True if request.content_type == "application/json" else False
     # TODO rate limit this view per caller ip?
-    if not is_valid_hostname:
-        return redirect("/")
-    if poll_node(host):
-        # Success!
+    if is_valid_hostname and poll_node(host):
+        if json:
+            return JsonResponse({'error': None})
```

Comment:
I would maybe prefer `result: success` and then `result: error, reason: foobar` but this is ok too :+1:

Reply:
oh I can change it!